src/main/java/com/tstepnik/planner/service/StatisticsService.java

```diff
@@ -0,0 +1,65 @@
+package com.tstepnik.planner.service;
+
+import com.tstepnik.planner.domain.Statistics;
+import com.tstepnik.planner.domain.user.User;
+import com.tstepnik.planner.repository.StatisticsRepository;
+import com.tstepnik.planner.repository.TaskRepository;
+import org.springframework.stereotype.Service;
+
+import java.text.DecimalFormat;
+import java.time.LocalDateTime;
+
+@Service
+public class StatisticsService {
+
+    private final TaskRepository taskRepository;
+    private final StatisticsRepository statisticsRepository;
+    private final TaskService taskService;
+    private final AuthService authService;
+
+    public StatisticsService(TaskRepository taskRepository, StatisticsRepository statisticsRepository,
+                             TaskService taskService, AuthService authService) {
+        this.taskRepository = taskRepository;
+        this.statisticsRepository = statisticsRepository;
+        this.taskService = taskService;
+        this.authService = authService;
+    }
+
+    public Statistics createSaveAndReturnStatistics() {
+        User loggedUser = authService.getLoggedUser();
+        Statistics statistics = statisticsRepository.findFirstByUserIdOrderByIdDesc(loggedUser.getId());
+        if (statistics == null || todayWasNotSavedStatistics(statistics.getCreationDate())) {
+            LocalDateTime now = LocalDateTime.now();
+            DecimalFormat df = new DecimalFormat("##.#");
+            Integer userArchivedTasks = countArchivedTasks();
+            Integer userFinishedTasks = countFinishTasks();
+            Double userProductivity = ((double) userFinishedTasks / userArchivedTasks) * 100;
+            String format = df.format(userProductivity).replaceAll(",", ".");
```

Comment:
This seems a little bit too much to just fomrat double properly. I am not sure but don't think we need to convert double to string and then string back to double.

Reply:
I did it for round decimal numbers.